include/nbl/builtin/hlsl/cpp_compat/intrinsics.hlsl

```diff
@@ -243,6 +243,74 @@ inline FrexpOutput<T> frexpStruct(NBL_CONST_REF_ARG(T) val)
 	return cpp_compat_intrinsics_impl::frexpStruct_helper<T>::__call(val);
 }
 
+#ifdef __HLSL_VERSION
+#define NAMESPACE spirv
+#else
+#define NAMESPACE glm
+#endif
+
+inline int32_t packSnorm4x8(float32_t4 vec)
+{
+	return NAMESPACE::packSnorm4x8(vec);
+}
+
+inline int32_t packUnorm4x8(float32_t4 vec)
+{
+	return NAMESPACE::packUnorm4x8(vec);
+}
+
+inline int32_t packSnorm2x16(float32_t2 vec)
+{
+	return NAMESPACE::packSnorm2x16(vec);
+}
+
+inline int32_t packUnorm2x16(float32_t2 vec)
+{
+	return NAMESPACE::packUnorm2x16(vec);
+}
+
+inline int32_t packHalf2x16(float32_t2 vec)
+{
+	return NAMESPACE::packHalf2x16(vec);
+}
+
+inline float64_t packDouble2x32(int32_t2 vec)
+{
+	return NAMESPACE::packDouble2x32(vec);
+}
+
+inline float32_t2 unpackSnorm2x16(int32_t val)
+{
+	return NAMESPACE::unpackSnorm2x16(val);
+}
+
+inline float32_t2 unpackUnorm2x16(int32_t val)
+{
+	return NAMESPACE::unpackUnorm2x16(val);
+}
+
+inline float32_t2 unpackHalf2x16(int32_t val)
+{
+	return NAMESPACE::unpackHalf2x16(val);
+}
+
+inline float32_t4 unpackSnorm4x8(int32_t val)
+{
+	return NAMESPACE::unpackSnorm4x8(val);
+}
+
+inline float32_t4 unpackUnorm4x8(int32_t val)
+{
+	return NAMESPACE::unpackUnorm4x8(val);
+}
+
+inline int32_t2 unpackDouble2x32(float64_t val)
```

Comment:
you may want to template with enable_if to stop implicit conversions that HLSL loves to do some much

Reply:
https://github.com/Devsh-Graphics-Programming/Nabla/blob/cpp_compat_intrinsic_functions/include/nbl/builtin/hlsl/cpp_compat/intrinsics.hlsl#L240

Reply:
like this?

Reply:
ye its fine.